### PR TITLE
IOS-3069 Add sync status loading animation

### PR DESCRIPTION
**Pull request by ignatovv**
https://github.com/anyproto/anytype-swift/assets/6589348/8419b936-a3fc-4725-b2c6-9f53760bb488

### CODE REVIEW

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/SyncStatusInfoNetworkIcon.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import SwiftUI
+import Services
+
+
+struct SyncStatusInfoNetworkIcon: View {
+    @State private var color: Color
+    @ObservedObject var model: SyncStatusInfoViewModel
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/SyncStatusInfoNetworkIcon.swift:7]: Really needed provide parent model?

Comment (mgolovko) [reply Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/SyncStatusInfoNetworkIcon.swift:7]: You can use `syncStatusInfo`

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/Model/Optional+SyncStatusInfo.swift [hunk 2]
```diff
@@ -29,7 +29,7 @@ extension Optional where Wrapped == SyncStatusInfo {
         }
     }
     
-    var networkIconColor: Color {
+    var networkIconColor: SyncStatusInfo.NetworkIconBackground {
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/Model/Optional+SyncStatusInfo.swift:32]: Extension for `extension Optional where Wrapped == SyncStatusInfo` requires  to sync types, methods, and mades a "hidden" logic. If you make a one default `SyncStatusInfo`,  won't it be more obvious?

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/SyncStatusInfoNetworkIcon.swift (hunk 1 above)
Comment (ignatovv) [reply Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/SyncStatusInfoNetworkIcon.swift:7]: Updated in this pr
https://github.com/anyproto/anytype-swift/pull/1781

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/Model/Optional+SyncStatusInfo.swift (hunk 2 above)
Comment (ignatovv) [reply Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/Model/Optional+SyncStatusInfo.swift:32]: Fixed in this pr
https://github.com/anyproto/anytype-swift/pull/1781/files#diff-d850f457cfe080deb43b173bba206055a58fd9fa7e6b9cebe1f2fd2081607520

**Review by mgolovko — APPROVED** [review]

**Pull request merged**